Swaylock fails to build on Fedora 35 (Rawhide)

Building swaylock 1.5 on Fedora 35 (Rawhide) 
```
In function ‘load_config’,
    inlined from ‘main’ at ../main.c:1140:23:
../main.c:1062:25: error: ‘free’ called on pointer returned from a mismatched allocation function [-Werror=mismatched-dealloc]
 1062 |                         free(config);
      |                         ^~~~~~~~~~~~
../main.c: In function ‘main’:
../main.c:1037:24: note: returned from ‘*fopen64’
 1037 |         FILE *config = fopen(path, "r");
      |                        ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

```
Werror is set so it stops the build.

--- a/main.c
+++ b/main.c
@@ -1059,7 +1059,7 @@ static int load_config(char *path, struct swaylock_state *state,
 		char *flag = malloc(nread + 3);
 		if (flag == NULL) {
 			free(line);
-			free(config);
+			fclose(config);
 			swaylock_log(LOG_ERROR, "Failed to allocate memory");
 			return 0;
 		}

